pkg/keyctl/key.go

```diff
@@ -62,3 +62,12 @@ func (k *Key) Unlink() error {
 	_, err := unix.KeyctlInt(unix.KEYCTL_UNLINK, int(k.id), int(k.ring), 0, 0)
 	return err
 }
+
+// Describe returns a string describing the attributes of a specified key
+func Describe(kID ID) (string, error) {
```

Comment:
No more public functions in this subpackage, please; we don’t want to maintain it as a stable public API.  Make them private in `pkg/docker/config`.

Reply:
Maybe we could (should?) move this package into `./internal` before the next major bump?

Reply:
why Unlink() can stay in pkg/keyctl/key.go but not Describe(kID ID)..
I think move keyctk package to `./internal` is better?

Reply:
The `pkg/keyctl` package was present in a release, so we can’t remove it, or any public functions/methods/types in it, without a major version bump.

Still, it’s not something users are expected to look for in c/image, and we don’t really want  to maintain or enhance it for such hypothetical external users — after all, it’s just a pretty small façade for `golang.org/x/sys/unix` now.

https://github.com/containers/image/pull/676 has already marked `pkg/keyctl` as deprecated: at the time, all callers could use either `golang.org/x/sys/unix` or `github.com/jsipprell/keyctl` without involving c/image.

This is a bit less true now that `ReadUserKeyring` seems to have no direct equivalent in either of those packages; still, c/image is not the obvious place to maintain that as a public utility.

Having the new code directly in `pkg/docker/config` (or maybe `pkg/docker/config/internal` or something like  `pkg/docker/config/internal/keyring`) would work well enough, I think; nothing else in the repo needs to care.

Before v4.0.0 we can then either moe the current `pkg/keyctl` to that same location, or just discard it replace its users with direct calls to `golang.org/x/sys/unix`.

Reply:
I added the new code to pkg/docker/config_linux.go, should I move it to pkg/docker/config.go?
Do I need to add test `readUserKeyring` in  pkg/docker/config/config_test.go?

Reply:
`removeAllAuthFromKernelKeyring` is Linux-specific, so it needs to be in a `*_linux.go` file (or a differently-named file that uses an explicit build tag).

Both `config_linux.go` and a separate `keyring_linux.go` would be fine with me.

If unit tests exist, having them (again, in a Linux-specific file) in the package would be nice, sure. (If they don’t, it’s not quite blocking the PR … but this is not quite trivial code, so tests would be nice.)